### PR TITLE
Setup initial benchmarks for model training

### PR DESCRIPTION
**Pull request by ianstenbit**
This adds benchmarks for only two models for now because I want to make sure parameterizing the benchmarks works without adding dozens of benchmarks at once.

This is a simple training benchmark which loads MNIST via TFDS (works on Borg) and trains for one epoch, recording training time, train accuracy, and compilation time.

Once this is merged, I will update our internal configs to automatically pick up these benchmarks daily.

### CODE REVIEW

File: keras_cv/benchmarks/classification_training_benchmark_test.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        metrics = []
+        metrics.append({"name": "compile_time", "value": compile_time})
+        metrics.append({"name": "avg_epoch_time", "value": training_time / self.epochs})
+        metrics.append({"name": "epochs", "value": self.epochs})
+        metrics.append(
+            {"name": "accuracy", "value": training_results.history["accuracy"][0]}
+        )
+
+        self.report_benchmark(wall_time=total_time, metrics=metrics)
```
Comment (tanzhenyu) [inline keras_cv/benchmarks/classification_training_benchmark_test.py:93]: out of curiosity -- will this fail if new benchmark is `worse` than previous benchmark? How do we set the tolerance diff?

File: keras_cv/benchmarks/classification_training_benchmark_test.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        self.strategy = tf.distribute.MirroredStrategy()
+
+    def benchmark_classification_training(self, app):
+        with self.strategy.scope():
+            t0 = time.time()
+
+            model = app(
+                include_top=True,
+                classes=self.num_classes,
+                input_shape=(56, 56, 1),
```
Comment (LukeWood) [inline keras_cv/benchmarks/classification_training_benchmark_test.py:64]: Am I wrong in thinking MNIST is 28x28?

**Review by LukeWood — COMMENTED** [review]
All in all looks good to me structurally.

File: keras_cv/benchmarks/classification_training_benchmark_test.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/benchmarks/classification_training_benchmark_test.py:93]: The tolerances are all configured internally with our benchmark running config. I pinged you a link

File: keras_cv/benchmarks/classification_training_benchmark_test.py (hunk 2 above)
Comment (LukeWood) [reply keras_cv/benchmarks/classification_training_benchmark_test.py:64]: Gotcha, good to know.  Thanks

**Review by LukeWood — APPROVED** [review]

**Review by tanzhenyu — APPROVED** [review]

**Pull request merged**